build.sbt

```diff
@@ -35,7 +35,14 @@ val sharedSettings = Project.defaultSettings ++ scalariformSettings ++  Seq(
 
   javacOptions ++= Seq("-target", "1.6", "-source", "1.6"),
 
-  scalacOptions ++= Seq("-unchecked", "-deprecation", "-optimize", "-Xlint", "-language:implicitConversions", "-language:higherKinds", "-language:existentials"),
+  scalacOptions ++= Seq(
+    "-unchecked",
+    "-deprecation",
+    "-optimize",
+    "-Xlint:-adapted-args",
```

Comment:
it is `Right(1, 2)` which is adapted to be `Right((1, 2))` because `Right(t: T)` takes one arg. This code is littered with that and it makes warnings totally overwhelm the errors (so I couldn't even find the errors).

It does not work on 2.10, so I'll roll that back, but we really need to clean this up.